docs/readmes/proposals/p006_hook_terravm_to_ci.md

```diff
@@ -0,0 +1,247 @@
+be able to access all the TipLab Infra
+
+2. CircleCI node can access the TipLab Infra and will update both agw and feg
+with the latest available tag using the FabFile (in the future those packages
+can be build directly on the CI node and transferred to the components directly
+or uploaded to a special TeraVM registry)
+
+3. CircleCI node can access the TipLab Infra and will use the fabfile to Trigger
+ng40 tests.
+
+##### Pros
+
+- Easy to implement
+
+##### Cons
+
+- CircleCI will move to FreedomFi which means we'll have to share vpn secrets
+with them.
+- OpenVPN server is running on an infra that we don't control
+
+
+
+#### Design 2
+
++-----------------------------------------------------+
+|  CircleCI Cloud                                     |
+|                                                     |
+|                                                     |
+|  +------------+               +-------------------+ |
+|  |            |               |                   | |
+|  |            |1. Spin node   |                   | |
+|  |CircleCI    +-------------->+CircleCI node      | |
+|  |            |  with secret  |                   | |
+|  |            |               |                   | |
+|  +------------+         +-----+-------------------+ |
+|                         |                           |
+|                         |                           |
++-----------------------------------------------------+
+                          |
++-----------------------------------------------------+
+|                         |                           |
+|  AWS/Packet             | 2. Call API service       |
+|                         | with token                |
+|        +----------------v--------------------+      |
+|        | TipLab Bastion                      |      |
+|        | +----------+   +---------+  +-----+ |      |
+|        | | Api      |   |OpenVPN  |  | logs| |      |
+|        | | SERVICE  |   |SERVICE  |  |     | |      |
+|        | +----------+   +---------+  +-----+ |      |
+|        +-------------------------------------+      |
++-----------------------------------------------------+
+               |            |                |
++-----------------------------------------------------+
+|              |3.update    | 3.update       | 4.Trigger
+|  TipLab Infra|            |                | tests  |
+|              |            |                |        |
+|  +-----------v+  +--------v-----+   +------v------+ |
+|  |            |  |              |   |             | |
+|  |            |  |              |   |             | |
+|  |  AGW       |  |   FEG        |   |   NG40      | |
+|  |            |  |              |   |             | |
+|  |            |  |              |   |             | |
+|  +------------+  +--------------+   +-------------+ |
+|                                                     |
++-----------------------------------------------------+
+
```

Comment:
in design 2, how will we receive the reports back from teravm?

Reply:
phase 1-4 is to just trigger it, nice reporting will occur in phase 5